statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/Translator.java

```diff
@@ -1,42 +1,74 @@
-                    .withCloudWatchLogsLogGroup(new CloudWatchLogsLogGroup()
+                destinations.add(new com.amazonaws.services.stepfunctions.model.LogDestination()
+                    .withCloudWatchLogsLogGroup(new com.amazonaws.services.stepfunctions.model.CloudWatchLogsLogGroup()
                             .withLogGroupArn(logGroup))
                 );
             }
-            result.setDestinations(destinations);
+            sfnLoggingConfiguration.setDestinations(destinations);
         }
 
-        return result;
+        return sfnLoggingConfiguration;
     }
 
-    public static com.amazonaws.services.stepfunctions.model.TracingConfiguration getTracingConfiguration(TracingConfiguration input) {
+    public static LoggingConfiguration getLoggingConfiguration(com.amazonaws.services.stepfunctions.model.LoggingConfiguration sfnLoggingConfiguration) {
+        LoggingConfiguration cfnLoggingConfiguration = new LoggingConfiguration();
 
-        com.amazonaws.services.stepfunctions.model.TracingConfiguration result = new com.amazonaws.services.stepfunctions.model.TracingConfiguration();
-        result.setEnabled(input.getEnabled());
+        cfnLoggingConfiguration.setLevel(sfnLoggingConfiguration.getLevel());
+        cfnLoggingConfiguration.setIncludeExecutionData(sfnLoggingConfiguration.getIncludeExecutionData());
 
-        return result;
+        if (sfnLoggingConfiguration.getDestinations() != null) {
+            List<LogDestination> destinations = sfnLoggingConfiguration.getDestinations().stream()
+                    .map(com.amazonaws.services.stepfunctions.model.LogDestination::getCloudWatchLogsLogGroup)
+                    .map(com.amazonaws.services.stepfunctions.model.CloudWatchLogsLogGroup::getLogGroupArn)
+                    .map(logGroupArn -> new LogDestination(new CloudWatchLogsLogGroup(logGroupArn)))
+                    .collect(Collectors.toList());
+
+            cfnLoggingConfiguration.setDestinations(destinations);
+        }
+
+        return cfnLoggingConfiguration;
+    }
+
+    public static com.amazonaws.services.stepfunctions.model.TracingConfiguration getTracingConfiguration(TracingConfiguration cfnTracingConfiguration) {
+        com.amazonaws.services.stepfunctions.model.TracingConfiguration sfnTracingConfiguration = new com.amazonaws.services.stepfunctions.model.TracingConfiguration();
+        sfnTracingConfiguration.setEnabled(cfnTracingConfiguration.getEnabled());
+
+        return sfnTracingConfiguration;
+    }
+
+    public static TracingConfiguration getTracingConfiguration(com.amazonaws.services.stepfunctions.model.TracingConfiguration sfnTracingConfiguration) {
+        TracingConfiguration cfnTracingConfiguration = new TracingConfiguration();
+        cfnTracingConfiguration.setEnabled(sfnTracingConfiguration.getEnabled());
+
+        return cfnTracingConfiguration;
+    }
+
+    public static List<TagsEntry> getTagsEntries(List<Tag> stateMachineTags) {
```

Comment:
Use `final` keyword for parameters which are not and should not be reassigned

Reply:
Updated